pkg/apis/pipeline/v1alpha1/merge.go

```diff
@@ -84,7 +84,8 @@ func MergeStepsWithStepTemplate(template *v1.Container, steps []Step) ([]Step, e
 			merged.Args = []string{}
 		}
 
-		steps[i] = Step{Container: *merged}
+		// Pass through original step Script, for later conversion.
+		steps[i] = Step{Container: *merged, Script: s.Script}
```

Comment:
coming in late to the party, but it looks like we have  `merge_test.go`, probs makes sense to add a test case for this?